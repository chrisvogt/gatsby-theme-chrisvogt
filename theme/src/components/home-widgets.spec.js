import React from 'react'
import renderer from 'react-test-renderer'
import HomeWidgets from './home-widgets'

import GitHub from '../components/widgets/github'
import RecentPosts from '../components/widgets/recent-posts'
import Spotify from '../components/widgets/spotify'

jest.mock('../components/widgets/github')
jest.mock('../components/widgets/recent-posts')
jest.mock('../components/widgets/spotify')

const getMockWidgetComponent = id => <div className={`MOCK__${id}Widget`}></div>

describe('HomeWidgets', () => {
  GitHub.mockImplementation(() => getMockWidgetComponent('GitHub'))
  RecentPosts.mockImplementation(() => getMockWidgetComponent('RecentPosts'))
  Spotify.mockImplementation(() => getMockWidgetComponent('Spotify'))

  it('matches the snapshot', () => {
    const tree = renderer.create(<HomeWidgets />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('widgets', () => {
    let testInstance

    beforeAll(() => {
      const testRenderer = renderer.create(<HomeWidgets />)
      testInstance = testRenderer.root
    })

    it('renders the GitHub widget', () => {
      expect(
        testInstance.findByProps({ className: 'MOCK__GitHubWidget' }).type
      ).toBe('div')
    })

    it('renders the Recent Posts widget', () => {
      expect(
        testInstance.findByProps({ className: 'MOCK__RecentPostsWidget' }).type
      ).toBe('div')
    })

    it('renders the Spotify widget', () => {
      expect(
        testInstance.findByProps({ className: 'MOCK__SpotifyWidget' }).type
      ).toBe('div')
    })
  })
})
