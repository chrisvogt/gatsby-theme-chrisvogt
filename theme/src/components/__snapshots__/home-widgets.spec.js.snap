// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeWidgets matches the snapshot 1`] = `
Array [
  <div
    className="MOCK__RecentPostsWidget"
  />,
  <div
    className="MOCK__GitHubWidget"
  />,
  <div
    className="MOCK__GoodreadsWidget"
  />,
  <div
    className="MOCK__SpotifyWidget"
  />,
]
`;
